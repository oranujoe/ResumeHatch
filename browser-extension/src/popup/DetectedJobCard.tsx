import React, { useState } from 'react';
import { JobPosting, ExtensionMessage } from '../messages';

interface Props {
  job: JobPosting;
  tabId: number;
}

const DetectedJobCard: React.FC<Props> = ({ job, tabId }) => {
  const handleSend = () => {
    const msg: ExtensionMessage = { type: 'OPEN_RESUME_HATCH', tabId };
    chrome.runtime.sendMessage(msg);
    window.close();
  };

  const [copied, setCopied] = useState(false);

  const handleCopy = () => {
    navigator.clipboard.writeText(job.description || '').then(() => {
      // Try native notification first
      const msg: ExtensionMessage = {
        type: 'SHOW_NOTIFICATION',
        title: 'Copied',
        message: 'Job description copied to clipboard.',
      };
      chrome.runtime.sendMessage(msg);

      // Always show inline feedback as a fallback
      setCopied(true);
      setTimeout(() => setCopied(false), 2000);
    });
  };

  return (
    <div className="space-y-2">
      <div className="border-l-4 border-blue-600 rounded-md p-3 bg-white shadow-md dark:bg-zinc-800">
        <p className="font-medium leading-snug line-clamp-2" title={job.title}>{job.title}</p>
        {job.company && (
          <p className="text-sm text-muted-foreground truncate" title={job.company}>{job.company}</p>
        )}
        {job.location && (
          <p className="text-xs text-muted-foreground truncate" title={job.location}>{job.location}</p>
        )}
      </div>
      <button onClick={handleSend} className="w-full rounded bg-blue-600 hover:bg-blue-700 active:bg-blue-800 text-white px-3 py-2 text-sm focus:outline-none focus:ring focus:ring-offset-2 focus:ring-blue-500 transition-colors">Send to Resume Hatch</button>
      <button onClick={handleCopy} className="w-full rounded border border-blue-600 text-blue-600 hover:bg-blue-50 dark:hover:bg-blue-900/20 px-3 py-2 text-sm focus:outline-none focus:ring focus:ring-offset-2 focus:ring-blue-500 transition">Copy description</button>
      {copied && (
        <p className="text-xs text-green-600 dark:text-green-400">Copied to clipboard</p>
      )}
    </div>
  );
};

export default DetectedJobCard; 