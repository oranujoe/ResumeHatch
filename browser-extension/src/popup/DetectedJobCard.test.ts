import { describe, it, expect, beforeAll } from 'vitest';
import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import DetectedJobCard from './DetectedJobCard';

// Mock minimal chrome API used by component
// @ts-ignore
(global as any).chrome = {
  runtime: {
    sendMessage: () => {},
    getURL: (path: string) => path,
  },
} as any;

// Mock clipboard API
// @ts-ignore
(global as any).navigator.clipboard = {
  writeText: () => Promise.resolve(),
};

const sampleJob = {
  title: 'AI Prompt Engineer',
  company: 'OpenAI',
  location: 'Remote',
  description: 'We are seeking a highly skilled prompt engineer to join our team...',
  url: 'https://linkedin.com/jobs/view/123',
};

describe('DetectedJobCard snapshot', () => {
  it('matches snapshot', () => {
    const element = React.createElement(DetectedJobCard, { job: sampleJob as any, tabId: 1 });
    const html = renderToStaticMarkup(element);
    expect(html).toMatchSnapshot();
  });
}); 