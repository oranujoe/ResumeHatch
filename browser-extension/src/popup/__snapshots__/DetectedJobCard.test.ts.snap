// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetectedJobCard snapshot > matches snapshot 1`] = `"<div class="space-y-2"><div class="border-l-4 border-blue-600 rounded-md p-3 bg-white shadow-md dark:bg-zinc-800"><p class="font-medium leading-snug line-clamp-2" title="AI Prompt Engineer">AI Prompt Engineer</p><p class="text-sm text-muted-foreground truncate" title="OpenAI">OpenAI</p><p class="text-xs text-muted-foreground truncate" title="Remote">Remote</p></div><button class="w-full rounded bg-blue-600 hover:bg-blue-700 active:bg-blue-800 text-white px-3 py-2 text-sm focus:outline-none focus:ring focus:ring-offset-2 focus:ring-blue-500 transition-colors">Send to Resume Hatch</button><button class="w-full rounded border border-blue-600 text-blue-600 hover:bg-blue-50 dark:hover:bg-blue-900/20 px-3 py-2 text-sm focus:outline-none focus:ring focus:ring-offset-2 focus:ring-blue-500 transition">Copy description</button></div>"`;
